app/src/lib/stores/app-store.ts

```diff
@@ -1024,17 +1024,46 @@ export class AppStore {
   }
 
   /** Update the menu with the names of the user's preferred apps. */
-  private updatePreferredAppMenuItemLabels() {
+  private async updatePreferredAppMenuItemLabels(repository?: Repository) {
```

Comment:
I think we should rename this method to something more general than being about "preferred apps".